db/db_impl/db_impl_compaction_flush.cc

```diff
@@ -119,13 +119,12 @@ Status DBImpl::SyncClosedLogs(JobContext* job_context) {
       s = directories_.GetWalDir()->Fsync();
     }
 
-    mutex_.Lock();
+    log_write_mutex_.Lock();
 
     // "number <= current_log_number - 1" is equivalent to
     // "number < current_log_number".
     MarkLogsSynced(current_log_number - 1, true, s);
     if (!s.ok()) {
-      error_handler_.SetBGError(s, BackgroundErrorReason::kFlush);
```

Comment:
Where do you set this error now?

Reply:
It will be saved at "db/db_impl/db_impl_compaction_flush.cc" L549 and L210

Reply:
In this function, we will not hold mutex so that we can not save BG error here.